python/ray/train/tests/test_lightning_deepspeed.py

```diff
@@ -0,0 +1,106 @@
+import os
+import pytest
+
+import ray
+from ray.air.config import CheckpointConfig, RunConfig
+from ray.air.constants import MODEL_KEY
+from ray.train.lightning import LightningConfigBuilder, LightningTrainer
+from ray.train.tests.lightning_test_utils import (
+    LinearModule,
+    DummyDataModule,
+)
+
+
+@pytest.fixture
+def ray_start_6_cpus_4_gpus():
+    address_info = ray.init(num_cpus=6, num_gpus=4)
+    yield address_info
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+@pytest.mark.parametrize("stage", [1, 2, 3])
+@pytest.mark.parametrize("test_restore", [True, False])
+def test_deepspeed_stages(ray_start_6_cpus_4_gpus, tmpdir, stage, test_restore):
+    # Reduce tests number
+    if test_restore and stage < 3:
+        return
```

Comment:
Restoring works for stage 3? Maybe modify this comment to be a TODO to test stage 3 + restore once it's possible.

Reply:
I sort of forgot why it didn't work for stage 3. 
Just to clear our minds. It works for driver storage, but not for cloud storage. It's because each worker won't download checkpoint from the cloud on restore?

Reply:
Yeah, there's some issues with downloading the latest checkpoint from cloud if a local copy exists.

Reply:
https://github.com/ray-project/ray/issues/36357

Reply:
Ok. Got it!